pkg/catalog/catalog.go

```diff
@@ -2738,6 +2738,10 @@ func (c *Catalog) CopyEntry(ctx context.Context, srcRepository, srcRef, srcPath,
 		if err != nil {
 			return nil, err
 		}
+
+		if srcRepo.StorageID != destRepo.StorageID {
+			return nil, fmt.Errorf("copy between different blockstores is not allowed: %w", graveler.ErrInvalidValue)
```

Comment:
```suggestion
			return nil, fmt.Errorf("%w: cannot copy between different blockstores", graveler.ErrInvalidValue)
```
But also note that this error message is quite nasty.  Consider using ErrInvalidStorageId, and also mentioning the two storage IDs.